internal/apmlog/logger.go

```diff
@@ -65,85 +80,98 @@ func initDefaultLogger() {
 		logWriter = &syncFile{File: f}
 	}
 
-	logLevel := errorLevel
-	if levelStr := strings.TrimSpace(os.Getenv("ELASTIC_APM_LOG_LEVEL")); levelStr != "" {
-		level, err := parseLogLevel(levelStr)
+	logLevel := DefaultLevel
+	if levelStr := strings.TrimSpace(os.Getenv(EnvLogLevel)); levelStr != "" {
+		level, err := ParseLogLevel(levelStr)
 		if err != nil {
-			log.Printf("invalid ELASTIC_APM_LOG_LEVEL %q, falling back to %q", levelStr, logLevel)
+			log.Printf("invalid %s %q, falling back to %q", EnvLogLevel, levelStr, logLevel)
 		} else {
 			logLevel = level
 		}
 	}
-	DefaultLogger = levelLogger{w: logWriter, level: logLevel}
+	DefaultLogger = &LevelLogger{w: logWriter, level: logLevel}
 }
 
+// Log levels.
 const (
-	debugLevel logLevel = iota
-	infoLevel
-	warnLevel
-	errorLevel
-	noLevel
+	DebugLevel Level = iota
+	InfoLevel
+	WarnLevel
+	ErrorLevel
+	CriticalLevel
+	OffLevel
 )
 
-type logLevel uint8
+// Level represents a log level.
+type Level uint32
 
-func (l logLevel) String() string {
+func (l Level) String() string {
 	switch l {
-	case debugLevel:
+	case DebugLevel:
 		return "debug"
-	case infoLevel:
+	case InfoLevel:
 		return "info"
-	case warnLevel:
+	case WarnLevel:
 		return "warn"
-	case errorLevel:
+	case ErrorLevel:
 		return "error"
 	}
 	return ""
 }
 
-func parseLogLevel(s string) (logLevel, error) {
+// ParseLogLevel parses s as a log level.
+func ParseLogLevel(s string) (Level, error) {
 	switch strings.ToLower(s) {
 	case "debug":
-		return debugLevel, nil
+		return DebugLevel, nil
 	case "info":
-		return infoLevel, nil
+		return InfoLevel, nil
 	case "warn":
```

Comment:
(This is my first time looking at the go agent, and really reviewing much Go code at all, so feel free to ignore if I'm missing something obvious.)

The [spec](https://github.com/elastic/apm/blob/master/specs/agents/logging.md) uses "warning" rather than "warn". Is this function parsing level strings from central config?

Reply:
@trentm Thanks, nice pick up! You're quite right, I missed that. I'll open another PR to fix it.